src/bindings/python/src/pyopenvino/core/common.hpp

```diff
@@ -144,6 +144,48 @@ class ref_wrapper {
     }
 };
 
+// See #18388
```

Comment:
please, have a look https://github.com/openvinotoolkit/openvino/pull/22874#discussion_r1493295495

I suppose this should be implemented on C++ side, because it's not a Python API issue:
```c++
std::shared_ptr<ov::Model> runtime_model = nullptr;
{
    Core core;
    CompiledModel cm = core.compile_model(model, "CPU");
    runtime_model = cm.get_runtime_model();
}
// issue during destroying of runtime_model 
```